### ISSUE TITLE
Conflict with Advanced Achievement

### ISSUE THREAD

**Issue by Nan1t** [Nan1t/Custom-Skins-Manager#1]
**[Moved from GitLab issues]**

If install Advanced Achievement 6.1.0 and Custom Skins Manager 3.1. Both plugins won't work.

Error log:
```
[Server thread/ERROR]: Error occurred while enabling CustomSkinsManager v3.1 (Is it up to date?)
java.lang.VerifyError: Bad access to protected data in getfield
Exception Details:
  Location:
    org/h2/store/fs/FilePathMem.moveTo(Lorg/h2/store/fs/FilePath;Z)V @11: getfield
  Reason:
    Type 'java/util/TreeMap' (current frame, stack[1]) is not assignable to 'org/h2/store/fs/FilePathMem'
  Current Frame:
    bci: @11
    flags: { }
    locals: { 'org/h2/store/fs/FilePathMem', 'org/h2/store/fs/FilePath', integer, 'java/util/TreeMap' }
    stack: { 'java/lang/String' }
  Bytecode:
    0000000: b200 2859 4ec2 1c9a 0045 2bb4 0029 2ab4
    0000010: 0019 b600 2f9a 0037 b200 282b b400 29b6
    0000020: 0034 9900 2a12 3505 bd00 2b59 032a b400
    0000030: 1953 5904 bb00 3759 b700 382b b600 3c12
    0000040: 3eb6 0041 b600 4553 b800 4bbf 2ab7 001f
    0000050: 3a04 1904 2bb4 0029 b600 51b2 0028 2ab4
    0000060: 0019 b600 5557 b200 282b b400 2919 04b6
    0000070: 0059 572d c3a7 000a 3a05 2dc3 1905 bfb1
    0000080:                                        
  Exception Handler Table:
    bci [6, 117] => handler: 120
    bci [120, 124] => handler: 120
  Stackmap Table:
    append_frame(@76,Object[#77])
    same_locals_1_stack_item_frame(@120,Object[#91])
    chop_frame(@127,1)

	at java.lang.Class.forName0(Native Method) ~[?:?]
	at java.lang.Class.forName(Class.java:315) ~[?:?]
	at org.h2.store.fs.FilePath.registerDefaultProviders(FilePath.java:84) ~[?:?]
	at org.h2.store.fs.FilePath.get(FilePath.java:52) ~[?:?]
	at org.h2.store.fs.FileUtils.isAbsolute(FileUtils.java:101) ~[?:?]
	at org.h2.engine.ConnectionInfo.getName(ConnectionInfo.java:378) ~[?:?]
	at org.h2.engine.Engine.openSession(Engine.java:50) ~[?:?]
	at org.h2.engine.Engine.openSession(Engine.java:179) ~[?:?]
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157) ~[?:?]
	at org.h2.engine.Engine.createSession(Engine.java:140) ~[?:?]
	at org.h2.engine.Engine.createSession(Engine.java:28) ~[?:?]
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351) ~[?:?]
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124) ~[?:?]
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103) ~[?:?]
	at org.h2.Driver.connect(Driver.java:69) ~[?:?]
	at org.h2.jdbcx.JdbcDataSource.getJdbcConnection(JdbcDataSource.java:189) ~[?:?]
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:352) ~[?:?]
	at org.h2.jdbcx.JdbcDataSource.getPooledConnection(JdbcDataSource.java:384) ~[?:?]
	at org.h2.jdbcx.JdbcConnectionPool.getConnectionNow(JdbcConnectionPool.java:225) ~[?:?]
	at org.h2.jdbcx.JdbcConnectionPool.getConnection(JdbcConnectionPool.java:197) ~[?:?]
	at ru.csm.api.storage.database.H2Database.<init>(H2Database.java:16) ~[?:?]
	at ru.csm.bukkit.SpigotSkinsManager.setupDatabase(SpigotSkinsManager.java:184) ~[?:?]
	at ru.csm.bukkit.SpigotSkinsManager.onEnable(SpigotSkinsManager.java:86) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.15.2.jar:git-Paper-334]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.15.2.jar:git-Paper-334]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.15.2.jar:git-Paper-334]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugin(CraftServer.java:472) ~[patched_1.15.2.jar:git-Paper-334]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugins(CraftServer.java:386) ~[patched_1.15.2.jar:git-Paper-334]
	at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:488) ~[patched_1.15.2.jar:git-Paper-334]
	at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:298) ~[patched_1.15.2.jar:git-Paper-334]
	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:900) ~[patched_1.15.2.jar:git-Paper-334]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by Iaiao** [comment]
Looks like now both CSM and latest version of Advanced Achievements use h2 1.4.200
Advanced Achievements 6.2.1 (latest free) and 6.7.2 (from their github repo) work fine with CSM for me

There is just a warning that it uses CSM's instance of h2: `[21:48:05 WARN]: [AdvancedAchievements] Loaded class org.h2.Driver from CustomSkinsManager v3.6 which is not a depend, softdepend or loadbefore of this plugin.`

**Comment by Nan1t** [comment]
> Looks like now both CSM and latest version of Advanced Achievements use h2 1.4.200
> Advanced Achievements 6.2.1 (latest free) and 6.7.2 (from their github repo) work fine with CSM for me
> 
> There is just a warning that it uses CSM's instance of h2: `[21:48:05 WARN]: [AdvancedAchievements] Loaded class org.h2.Driver from CustomSkinsManager v3.6 which is not a depend, softdepend or loadbefore of this plugin.`

Hm, this is good news, thanks! Anyway, new dependency system i the next release should solve this problem, even if versions of H2 will be changed.

**Comment by Nan1t** [comment]
Fixed in CSM v3.6.
